### PR TITLE
Move workload latency metrics lifecycle to limiter

### PR DESCRIPTION
**Pull request by DariaKunoichi**
##### Checklist

- [x] Tested in playground or other setup
- [ ] Documentation is changed or added
- [x] Tests and/or benchmarks are included
- [ ] Breaking changes

### Description of change

- - -
This change is [Reviewable](https://reviewable.io/reviews/fluxninja/aperture/609)

### CODE REVIEW

File: pkg/policies/dataplane/actuators/rate/rate-limiter.go [hunk 1]
```diff
@@ -355,3 +355,9 @@ func (rateLimiter *rateLimiter) GetLimiterID() iface.LimiterID {
 		PolicyHash:     rateLimiter.GetPolicyHash(),
 	}
 }
+
+// GetHistogram is there to satisfy Limiter interface.
+func (rateLimiter *rateLimiter) GetHistogram(map[string]string) prometheus.Observer {
```
Comment (tanveergill) [inline pkg/policies/dataplane/actuators/rate/rate-limiter.go:360]: FYI, would need metrics for RateLimiter too: https://github.com/fluxninja/aperture/issues/459

At least a count metric will be required. Can do a latency summary as well

File: pkg/policies/dataplane/iface/limiter.go [hunk 2]
```diff
@@ -28,4 +29,5 @@ type Limiter interface {
 	GetSelector() *selectorv1.Selector
 	RunLimiter(labels map[string]string) *flowcontrolv1.LimiterDecision
 	GetLimiterID() LimiterID
+	GetHistogram(labels map[string]string) prometheus.Observer
```
Comment (tanveergill) [inline pkg/policies/dataplane/iface/limiter.go:32]: Its not really a Histogram but a Summary. [`prometheus.Observer`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Observer) is the common interface for both histograms and summary. Can rename accordingly.

**Review by tanveergill — APPROVED** [review]
Good to merge after addressing minor comments :+1:

File: pkg/policies/dataplane/actuators/rate/rate-limiter.go (hunk 1 above)
Comment (DariaKunoichi) [reply pkg/policies/dataplane/actuators/rate/rate-limiter.go:360]: Ok, I'll take this task from Hasit

File: pkg/policies/dataplane/iface/limiter.go (hunk 2 above)
Comment (DariaKunoichi) [reply pkg/policies/dataplane/iface/limiter.go:32]: Ok

**Pull request merged**